llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java

```diff
@@ -229,11 +230,20 @@ public void debugDumpShort(StringBuilder sb) {
         new LinkedBlockingQueue<Runnable>(),
         new ThreadFactoryBuilder().setNameFormat("IO-Elevator-Thread-%d").setDaemon(true).build());
     FixedSizedObjectPool<IoTrace> tracePool = IoTrace.createTracePool(conf);
+    if (isEncodeEnabled) {
+      int encodeThreads = numThreads * 2;
```

Comment:
Why exactly 2 times?

Reply:
Yeah quite arbitrary I'll give you that.
Text reading is actually started by the "regular" IO threads, and if encoding is needed for cache insertion, than one of these threads can produce more async encode tasks. This may end up being bursty, and according to what I've seen a bigger thread pool could come handy.
Anyway I made this configurable now.